lib/blockchain/transaction.rs

```diff
@@ -61,16 +56,14 @@ impl Transaction {
   /// HashSet of hashed inputs.
   pub fn input_hashes(&self) -> HashSet<String>{
     self.inputs.iter().map(|input| {
-      let input_bytes = self.output_bytes(input);
-      self.generate_hash(input_bytes)
+      self.generate_hash(input)
     }).collect::<HashSet<String>>()
   }
 
   /// HashSet of hashed outputs.
   pub fn output_hashes(&self) -> HashSet<String>{
     self.outputs.iter().map(|output| {
-      let output_bytes = self.output_bytes(output);
-      self.generate_hash(output_bytes)
+      self.generate_hash(output)
     }).collect::<HashSet<String>>()
```

Comment:
The code is fine but same problem as the `input/output_value` functions. Basically these functions have too much knowledge of the internal workings of `Output` and _force_ us into duplicating code (both are big smells!).

Same fix as above comment would be fine (or an even better solution ;)